crates/biome_js_analyze/src/lint/nursery/use_single_js_doc_asterisk.rs

```diff
@@ -0,0 +1,311 @@
+    is_end_line: bool,
+    line_index: usize,
+    char_start: usize,
+    char_end: usize,
+}
+
+fn char_is_whitespace(c: u8) -> bool {
+    c == b' ' || c == b'\t'
+}
+
+fn char_is_asterisk(c: u8) -> bool {
+    c == b'*'
+}
+
+fn get_first_asterisk_index(line: &str) -> usize {
+    line.as_bytes()
+        .iter()
+        .take_while(|&&b| !char_is_asterisk(b))
+        .count()
+        + 1
+}
+
+fn get_invalid_jsdoc_line_start(text: &str) -> Option<InvalidJsDocLineIndexes> {
+    let lines = text.lines();
+    let lines_count = lines.clone().count();
+
+    let (invalid_line_index, invalid_line_char_end) = lines
+        // Skip first/last lines
+        .take(lines_count - 1)
+        .enumerate()
+        .skip(1)
+        .find_map(|(line_index, line)| {
+            let line = line.trim_start();
+            let bytes = line.as_bytes();
+
+            // If first symbol is not an asterisk, we can skip the line
+            if bytes.first().is_some_and(|&b| !char_is_asterisk(b)) {
+                return None;
+            }
+
+            let mut invalid_asterisk_index = None;
+
+            let mut byte_it = bytes.iter().skip(1).enumerate().peekable();
+            while let Some((char_index, &b)) = byte_it.next() {
+                if char_is_asterisk(b) {
+                    // double asterisk is valid
+                    if byte_it
+                        .peek()
+                        .is_some_and(|&(_, &next_b)| char_is_asterisk(next_b))
+                    {
+                        return invalid_asterisk_index.map(|_| (line_index, char_index));
+                    }
+
+                    invalid_asterisk_index = Some(char_index);
+                    continue;
+                }
+
+                if char_is_whitespace(b) {
+                    continue;
+                }
+
+                // Found non-whitespace character
+                break;
+            }
+
+            invalid_asterisk_index.map(|char_index| (line_index, char_index + 1))
+        })?;
+
+    let start = get_first_asterisk_index(text.lines().nth(invalid_line_index)?);
+
+    Some(InvalidJsDocLineIndexes {
+        is_end_line: false,
+        line_index: invalid_line_index,
+        char_start: start,
+        char_end: start + invalid_line_char_end,
+    })
+}
+
+fn get_invalid_jsdoc_last_line(text: &str) -> Option<InvalidJsDocLineIndexes> {
+    let ending_marker = 2; // "*/"
+    let last_line = text.lines().last()?.trim_end();
+
+    let mut last_invalid_asterisk_index = None;
+
+    let byte_it = last_line
+        .as_bytes()
+        .iter()
+        .enumerate()
+        .rev()
+        .skip(ending_marker);
+
+    for (char_index, &b) in byte_it {
+        if char_is_asterisk(b) {
+            last_invalid_asterisk_index = Some(char_index);
+            continue;
+        }
+
+        if char_is_whitespace(b) {
+            continue;
+        }
+
+        // Found non-whitespace character
+        break;
```

Comment:
I think you can make your intention clearer like this:

```suggestion
        if !char_is_whitespace(b) {
            break;
        }
```